gstreamer-rtsp-server/src/auto/r_t_s_p_auth.rs

```diff
@@ -0,0 +1,170 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 9ae0a4a+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use ffi;
+use gio;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+#[cfg(any(feature = "v1_12", feature = "dox"))]
+use gst_rtsp;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct RTSPAuth(Object<ffi::GstRTSPAuth, ffi::GstRTSPAuthClass>);
+
+    match fn {
+        get_type => || ffi::gst_rtsp_auth_get_type(),
+    }
+}
+
+impl RTSPAuth {
+    pub fn new() -> RTSPAuth {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_rtsp_auth_new())
+        }
+    }
+
+    pub fn check(check: &str) -> bool {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib(ffi::gst_rtsp_auth_check(check.to_glib_none().0))
+        }
+    }
+
+    pub fn make_basic(user: &str, pass: &str) -> String {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_rtsp_auth_make_basic(user.to_glib_none().0, pass.to_glib_none().0))
+        }
+    }
+}
+
+impl Default for RTSPAuth {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+unsafe impl Send for RTSPAuth {}
+unsafe impl Sync for RTSPAuth {}
+
+pub trait RTSPAuthExt {
+    //fn add_basic(&self, basic: &str, token: /*Ignored*/&mut RTSPToken);
+
+    //#[cfg(any(feature = "v1_12", feature = "dox"))]
+    //fn add_digest(&self, user: &str, pass: &str, token: /*Ignored*/&mut RTSPToken);
+
+    //fn get_default_token(&self) -> /*Ignored*/Option<RTSPToken>;
+
+    #[cfg(any(feature = "v1_12", feature = "dox"))]
+    fn get_supported_methods(&self) -> gst_rtsp::RTSPAuthMethod;
+
+    //fn get_tls_authentication_mode(&self) -> /*Ignored*/gio::TlsAuthenticationMode;
+
+    fn get_tls_certificate(&self) -> Option<gio::TlsCertificate>;
+
+    //fn get_tls_database(&self) -> /*Ignored*/Option<gio::TlsDatabase>;
```

Comment:
Also this needs to be added at some point, together with the corresponding pieces in GIO